commit c70609e72f7eef422af122543d66c595ee139719
Author: Benda Xu <heroxbd@gentoo.org>
Date:   Mon Jun 20 09:12:35 2016 +0900

    let env-update call ldconfig if it exists
    
        In Prefix, let env-update call ldconfig if it exists.
    
        It does not affect prefix-rpath, and handles prefix-libc(RAP) correctly.

diff --git a/pym/portage/util/env_update.py b/pym/portage/util/env_update.py
index c0a93a8..fde2f66 100644
--- a/pym/portage/util/env_update.py
+++ b/pym/portage/util/env_update.py
@@ -306,13 +306,17 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env,
 		if not libdir_contents_changed:
 			makelinks = False
 
-	ldconfig = "/sbin/ldconfig"
 	if "CHOST" in settings and "CBUILD" in settings and \
 		settings["CHOST"] != settings["CBUILD"]:
 		ldconfig = find_binary("%s-ldconfig" % settings["CHOST"])
+	else:
+		ldconfig = os.path.join(eroot, "sbin", "ldconfig")
+
+	if not (os.access(ldconfig, os.X_OK) and os.path.isfile(ldconfig)):
+		ldconfig = None
 
 	# Only run ldconfig as needed
-	if makelinks and ldconfig and not eprefix:
+	if makelinks and ldconfig:
 		# ldconfig has very different behaviour between FreeBSD and Linux
 		if ostype == "Linux" or ostype.lower().endswith("gnu"):
 			# We can't update links if we haven't cleaned other versions first, as
